tensornetwork/backends/pytorch/pytorch_backend.py

```diff
@@ -85,7 +86,12 @@ def diag(self, tensor: Tensor) -> Tensor:
     return self.torch.diag(tensor)
 
   def convert_to_tensor(self, tensor: Tensor) -> Tensor:
-    return self.torch.as_tensor(tensor)
+    result = self.torch.as_tensor(tensor)
+    if self.dtype is not None and result.dtype is not self.dtype:
```

Comment:
Add this check before doing the conversion.

Reply:
Is there any advantage? The reason it's like that is that `tensor` can be a `np.ndarray` (usually is for me), but `self.dtype` is a `torch.dtype`. For example, `torch.float64`  compares `False` to `np.float64`, even though those `dtypes` are compatible

Reply:
Aw I see your point now. Ok, lets keep it as is then.